### ISSUE TITLE
Support for ATV and IPadOs

### ISSUE THREAD

**Issue by TheMBeat** [immichFrame/ImmichFrame#137]
### Description

Support Apple TV and IPadOs 

### Use Case

Slideshow on big TVs 

### Pre-Submission Checklist

- [X] This is a feature request and not a bug report

- [X] I have checked for related issues and checked the documentation

**Comment by 3rob3** [comment]
While this is technically pretty easy (to add iOS support), the Apple developer fee is $100 every year. I paid the Google fee since it’s only one time $25, but I just can’t justify ongoing Apple fees on a free app.

**Comment by dezerving** [comment]
> While this is technically pretty easy (to add iOS support), the Apple developer fee is $100 every year. I paid the Google fee since it’s only one time $25, but I just can’t justify ongoing Apple fees on a free app. 

Any thoughts to add iOS support, and then just allow people to sideload? Wouldn't need to pay every year.

**Comment by TheMBeat** [comment]
I can understand that. If the code is there, anyone can compile it themselves. Or donate it.

**Comment by JW-CH** [comment]
I/We are working on a hosted (web) version of immichFrame. Maybe this would be something to consider for you.

**Comment by JW-CH** [comment]
Closing this for now. You can try the web version. https://github.com/immichFrame/ImmichFrame/blob/main/Install_Web.md#-installation

**Comment by dezerving** [comment]
Don't really know why this is closed as _completed_ since the web version still isn't compatible with AppleTV because it requires you to side load an experimental browser.

From what's been said, the port shouldn't be all that complicated. The $100/year fee is not acceptable for a free app so I get it.

I feel like a good option could be to make the iOS/tvOS apps paid. I'm sure most would happily pay up to $2.99 for this. There's **a lot** of people on the official Immich repo wanting an AppleTV app and this could fill that gap for them.

If sales drop off after a year and aren't offsetting the $100 fee, you could remove the app and just maintain an ipa that can be sideloaded.

**Comment by 3rob3** [comment]
> Don't really know why this is closed as _completed_ since the web version still isn't compatible with AppleTV because it requires you to side load an experimental browser.
> 
> From what's been said, the port shouldn't be all that complicated. The $100/year fee is not acceptable for a free app so I get it.
> 
> I feel like a good option could be to make the iOS/tvOS apps paid. I'm sure most would happily pay up to $2.99 for this. There's **a lot** of people on the official Immich repo wanting an AppleTV app and this could fill that gap for them.
> 
> If sales drop off after a year and aren't offsetting the $100 fee, you could remove the app and just maintain an ipa that can be sideloaded.

All good points. I think it was closed thinking more about iOS than TV. 
My family is an all iOS household so I get it, imagine being me developing and maintaining Android when I don't even use it :). Personally, with respect to Apple TV, I lost interest when I discovered Apple doesn't let you use 3rd party screensavers. That is the coolest part for me with the TV apps.
Genuinely asking, you really are interested in an iOS app that when launched is a full screen "photo frame" cycling thru immich photos?

**Comment by dezerving** [comment]
> All good points. I think it was closed thinking more about iOS than TV. 
>
> My family is an all iOS household so I get it, imagine being me developing and maintaining Android when I don't even use it :). Personally, with respect to Apple TV, I lost interest when I discovered Apple doesn't let you use 3rd party screensavers. That is the coolest part for me with the TV apps.

I see now that makes sense. And same, we're an all iOS household as well. Yeah if we could use these as screensavers it would be awesome. That is disappointing.


> Genuinely asking, you really are interested in an iOS app that when launched is a full screen "photo frame" cycling thru immich photos?

Mainly tvOS, yeah. Would be nice to just load that up during a family get together and have it cycle through event photos or whatever. Currently, AirPlay mirroring is not that reliable. I don't see iOS as that useful.

A bit off topic but since this is available on Android are you using a tablet as a photo frame? If so what's the recommended hardware?

**Comment by kissu** [comment]
I understand the friction of developing for the Apple TV for sure.
But that would also be a killer feature 🤩  indeed, and I think that you could definitely offset it with a one-time payment of some sort or asking for some sponsorship. 👍🏻

**Comment by 3rob3** [comment]
> I understand the friction of developing for the Apple TV for sure. But that would also be a killer feature 🤩 indeed, and I think that you could definitely offset it with a one-time payment of some sort or asking for some sponsorship. 👍🏻
> 
> I come from here btw: [immich-app/immich#3892](https://github.com/immich-app/immich/discussions/3892)
> 
> You could sell it with a ["fair price"](https://jordanbaird.gumroad.com/l/ice?layout=profile&recommended_by=library) approach or just slap a 2€ premium on it on the AppleTV store. Given how many people are interested, they wouldn't mind paying that. This is also considering the fact that it's the price for a regular Web browser on the Apple TV store too. But I think that having a bit of control over the screen saver + other features might make it far more interesting to the audience here rather than paying for a browser + opening it from a local URL. 🙏🏻

Remember, you cannot set a 3rd party screensaver on Apple TV. So this would strictly be a launch-able slideshow app.

**Comment by kissu** [comment]
Yes, I very much assumed so haha. Would be very un-Apple otherwise.
(Also, even if we can side-load another store on iPhones nowadays, AppleTV will definitely not have such a nice treatment before long)
But still, fine with the community I think. You can launch the app and let it cycle. Not as good as it behaves to a screensaver but it's fine.
Idea would meanwhile be to not allow the native screensaver to kick in while ImmichFrame is running (quite feasible I think, some apps like `Blank` already prevent the native screensaver).
Also, I could still Airplay some music on external speakers while ImmichFrame is running, hence no need to even wire a Music Player to ImmichFrame. 👍🏻

**Comment by 3rob3** [comment]
I started some preliminary work on this. Will be awhile before its ready. I had a functioning iOS app, but Apple TV does not allow WebViews so I will have to re-do everything similar to how I did for Android Frameo devices that don't support WebView.

**Comment by kissu** [comment]
Amazing man, thanks a lot! 🤩 
Better to take your time and have something nice that works and people love than nothing at all, hence people probably don't mind waiting a little since there's no solution at all for such a feature as of right now. ❤

**Comment by dezerving** [comment]
> I started some preliminary work on this. Will be awhile before its ready. I had a functioning iOS app, but Apple TV does not allow WebViews so I will have to re-do everything similar to how I did for Android Frameo devices that don't support WebView.

Excited for this Rob, even with the AppleTV screensaver restrictions. Let me know if/how we can help test!

**Comment by 3rob3** [comment]
Made some good progress today :)

https://github.com/user-attachments/assets/85e2366c-5faa-4f2d-8114-5e08ab880216

**Comment by dezerving** [comment]
> Made some good progress today :)
> IMG_3507.mp4

This is awesome. Can't wait to try it out :D

**Comment by 3rob3** [comment]
App is in review with Apple.

**Comment by 3rob3** [comment]
Now available for Apple TV:
https://apps.apple.com/us/app/immichframe/id6742748077

For iOS and iPadOS you can create a fullscreen PWA app by opening the URL, and adding to homescreen

**Comment by 3rob3** [comment]
Arrow up should enter settings again. Is this not working?

**Comment by brandonb927** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/1509352?u=a40783e786589b9e6d154e1aa9cf5a1d4d1ea0bc&amp;v=4&amp;size=80">@brandonb927](https://github.com/brandonb927?rgh-link-date=2025-03-06T16%3A39%3A39.000Z) Arrow/directional button up should enter settings again. Is this not working?

Ah! Thank you, I was using the AppleTV app on my iPhone to set things up because of the keyboard, didn't even occur to me to use the physical remote 😅

**Comment by 3rob3** [comment]
Oh good. 🙂

**Comment by 3rob3** [comment]
You are putting the ImmichFrame server url, not the ImmichFrame url right? Die the burp work in browser?

**Comment by brandonb927** [comment]
> You are putting the ImmichFrame server url, not the Immich url right? Does the url work in browser?

🤦‍♂️🤦‍♂️ Please ignore my ineptitude; I will followup if I have an _actual_ issue that is no PEBKAC.

**Comment by 3rob3** [comment]
Haha, no problem!
